src/pkg/packager/composer/list.go

```diff
@@ -229,20 +229,24 @@ func (ic *ImportChain) Compose() (composed types.ZarfComponent, err error) {
 	}
 
 	// start with an empty component to compose into
-	composed = types.ZarfComponent{}
+	composed = &types.ZarfComponent{}
 
 	// start overriding with the tail node
 	node := ic.tail
 	for node != nil {
 		fixPaths(&node.ZarfComponent, node.relativeToHead)
 
 		// perform overrides here
-		overrideMetadata(&composed, node.ZarfComponent)
-		overrideDeprecated(&composed, node.ZarfComponent)
-		overrideResources(&composed, node.ZarfComponent)
-		overrideActions(&composed, node.ZarfComponent)
+		err := overrideMetadata(composed, node.ZarfComponent)
+		if err != nil {
+			return composed, err
```

Comment:
Would probably be good to update all error handling in `Compose()` to return a nil pointer

```suggestion
			return nil, err
```